src/redux/reducers/session.js

```diff
@@ -0,0 +1,17 @@
+import { LOG_IN_SUCCESS, LOG_OUT } from '../actionTypes';
+// import { browserHistory } from 'react-router';
+
+const INITIAL_STATE = {
+	session: !!sessionStorage.jwt
+}
+
+export default function sessionReducer(state = INITIAL_STATE, action) {
+  switch(action.type) {
+    case LOG_IN_SUCCESS:
+      return !!sessionStorage.jwt
+    case LOG_OUT:
+      return {...state, session: false}
```

Comment:
The value of the session is always a boolean, double exclamation transforms the value to true if the token exists of to false otherwise. Checking if the token is the correct one is on the backend though. The point here is that it checks if there is a token both after login or just page refresh. Is this ok? I guess I can create a function to check it on log in and also on document load, but the store does this very nicely without additional code.

Reply:
Better if we just revert back to Milestone 5 code @YellowPipe. I won't be checking anymore the parts where you included Milestone 6 features. So it would make it easier for me to review if you could provide a version that removes all the asynchronous code as well as coupling with the backend API. Thank you in advance!

Reply:
Let's focus first on accomplishing up to Milestone 5 then I promise you we can discuss the other features :) Important thing is to get the thumbs up first.